### PR TITLE
Screen Wake Lock: Add test for non-fully active doc

### PR DESCRIPTION
**Pull request by marcoscaceres**
Add test for https://github.com/w3c/screen-wake-lock/pull/367

### CODE REVIEW

**Review by rakuco — COMMENTED** [review]
The test looks fine, although it seems to be checking a condition that's already covered by the other ones.

I wonder if it'd be possible to add a test for the new check being added in w3c/screen-wake-lock#367 (i.e. using `window_state_context()` to change the page visibility after calling request() and verifying the latter fails).

**Review by reillyeon — APPROVED** [review]

File: screen-wake-lock/wakelock-active-document.https.window.js [hunk 1]
```diff
@@ -38,7 +38,27 @@ promise_test(async t => {
   );
   // We are done, so clean up.
   iframe.remove();
-}, "navigator.wakeLock.request() aborts if the document is not active.");
+}, "navigator.wakeLock.request() aborts if the document becomes not active.");
+
+promise_test(async t => {
+  const iframe = document.createElement("iframe");
+  document.body.appendChild(iframe);
+  // We first got to page1.html, grab a WakeLock object.
```
Comment (marcoscaceres) [inline screen-wake-lock/wakelock-active-document.https.window.js:46]:
```suggestion
```

**Comment by marcoscaceres** [comment]
> The test looks fine, although it seems to be checking a condition that's already covered by the other ones.

Yeah, you are right... it's a simplified version of the others as it's just a single iframe. I'm ok to keep it though as it could prove useful in case something goes wrong (easier to debug than the other ones). 

> I wonder if it'd be possible to add a test for the new check being added in https://github.com/w3c/screen-wake-lock/pull/367 (i.e. using window_state_context() to change the page visibility after calling request() and verifying the latter fails).

Yeah, that's a good idea. However, https://github.com/w3c/screen-wake-lock/pull/367 is just dealing with fully active stuff. We should have another test just for testing visibility. We should do that in separate file though, because otherwise (in webkit at least) we wouldn't be able to run this test on iOS devices, because we can't hide windows/tabs in automation on mobile.

**Comment by marcoscaceres** [comment]
Filed https://github.com/w3c/screen-wake-lock/issues/373

**Pull request merged**